### DISCUSSION TITLE
Is there a a SCPI command to delete a trace?

### DISCUSSION THREAD

**Discussion by AirEnt** [jankae/LibreVNA#278] [Q&A] [answered]
Hi. I am writing some software that interfaces to the VNA via the SCPI server. Part of this task is to automatically setup the traces. This means that sometimes I would like to delete traces. Is there a SCPI command that allows me to delete traces either by index or name? I cant seem to find anything in the SCPI Programming Guide.

**Comment by jankae** [comment]
Looks like I forgot to include such a command. Right now, there is no way of deleting traces via SCPI. I'll add this, should be quick and easy addition.

**Reply by fkutlu007** [reply]
> Looks like I forgot to include such a command. Right now, there is no way of deleting traces via SCPI. I'll add this, should be quick and easy addition.

Hi. I am also writing a software that interfaces to VNA via SCPI server in a similar way. When I was looking at this title, I was interested in your adding SCPI command. Can you give me information on how to do this? I would like to be able to add SCPI command.

**Reply by jankae** [reply]
What are you trying to do?
- if you want to delete traces via SCPI, this command is already available
- if you want to add some other command, this requires a change to the GUI code (written in C++/Qt). Depending on the command you want to implement, this can be an easy change requiring just a few lines of code or it can turn into a large and difficult project. This is the [commit](https://github.com/jankae/LibreVNA/commit/a5b9195ac95703350468dd428f11a4fd6ae29b5d#diff-4906259afb670d6317ca76401543d1a64016e33e93b939f52b43a6589d1f55aa) that added the :VNA:TRACe:DELete command, maybe it can help get you started.

**Reply by fkutlu007** [reply]
I want to add other commands. I will look at the commit you added. Thanks for this.

**Reply by jankae** [reply]
If you do end up adding new commands, I would appreciate a pull request so everyone can benefit. Alternatively, you can also just request new commands if you do not want to code them yourself (by creating a new issue). I can not promise anything but if it is fairly easy to implement, I will do that.

**Reply by fkutlu007** [reply]
Actually, I'm thinking of adding an SCPI command to save data in S2P format. I couldn't figure out how to do this with the existing commands. I want this to use while controlling the device with my own interface. This seems quite simple. I'd be happy if you consider adding it, as I think it would be useful for everyone. As I try different implementations, I'll provide updates so that everyone can use and improve it, of course.

**Reply by jankae** [reply]
You should take a look at the VNA:TRACe:TOUCHSTONE command. It is a bit of a clunky command because it does contain newlines in its response and there is no easy way to know when the full response has been parsed (other than a timeout or knowing the stop frequency and looking for that). But it might just do what you want.

**Reply by fkutlu007** [reply]
This will be very useful for me. 
If you have any suggestions on how to make serial measurements, I would like to hear them.

**Reply by jankae** [reply]
I am not sure what you mean by serial measurements. I am happy to help, but if it is not related to SCPI commands, opening a new discussion is probably better than derailing this one.

**Comment by jankae** [comment, marked as answer]
There is a new SCPI command now :VNA:TRACe:DELete

**Reply by AirEnt** [reply]
Thanks for this. It really helps.
